src/test/macros/macro_test.h

```diff
@@ -0,0 +1,19 @@
+#include <gtest/gtest.h>
+
+#include "track/macro.h"
+
+namespace {
+const QString kConfigGroup = QStringLiteral("[MacroRecording]");
+const QString kChannelGroup("[Channel1]");
+
+const MacroAction kAction(25'000, 7'500);
+/// Checks that a Macro containing the given action was successfully recorded,
+/// including the extra first loop action.
+void checkMacroAction(MacroPointer macro, MacroAction action = kAction) {
```

Comment:
```
In file included from /home/jan/Projects/mixxx/src/test/macros/macro_test.cpp:1:
/home/jan/Projects/mixxx/src/test/macros/macro_test.h:12:6: error: ‘void {anonymous}::checkMacroAction(MacroPointer, MacroAction)’ defined but not used [-Werror=unused-function]
   12 | void checkMacroAction(MacroPointer macro, MacroAction action = kAction) {
      |      ^~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/mixxx-test.dir/build.make:784: CMakeFiles/mixxx-test.dir/src/test/macros/macro_test.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/jan/Projects/mixxx/src/test/macros/macroplayback_test.cpp:1:
/home/jan/Projects/mixxx/src/test/macros/macro_test.h:12:6: error: ‘void {anonymous}::checkMacroAction(MacroPointer, MacroAction)’ defined but not used [-Werror=unused-function]
   12 | void checkMacroAction(MacroPointer macro, MacroAction action = kAction) {
      |      ^~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/mixxx-test.dir/build.make:771: CMakeFiles/mixxx-test.dir/src/test/macros/macroplayback_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1116: CMakeFiles/mixxx-test.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
```

Please always build with `-DWARNING_FATAL=ON` to catch these issues before committing.

Anonymous namespaces do not belong into header files. I suggest you make subclass BaseMacroTest subclass of MixxxTest and make these static members.

Reply:
I don't want a MixxxTest everywhere :) but I will find a solution

Reply:
ping.

Reply:
Sorry, pretty busy recently. But I still have this PR in the back of my head ;)

Reply:
Can't reproduce this warning/error with that flag, did something change?

Reply:
Cmake is a big yikes. It has many different commands that take different arguments but are differentiated only by slight syntax differences: https://cmake.org/cmake/help/latest/manual/cmake.1.html#generate-a-project-buildsystem

In the end I edited the `CMakeCache.txt` by hand and now have it as well ^^